hist/hist/src/HFitImpl.cxx

```diff
@@ -928,29 +932,25 @@ TFitResultPtr ROOT::Fit::UnBinFit(ROOT::Fit::UnBinData * data, TF1 * fitfunc, Fo
 
    }
 
-   // store result in the backward compatible VirtualFitter
-   // in case not running in a multi-thread enabled mode
-   if (gGlobalMutex) {
+   // store fitting result in the backward compatible TVirtualFitter object
+   // lock in case running in a multi-thread enabled mode
+   {
```

Comment:
TVirtualFitter::GetFitter is a legacy functionality and in general should not be used, in particular in an application running MT. We could add a note in the documentation of `TVirtualFitter::GetFitter` that it should not be used in MT. 

I will apply your suggestion to move later in the code the lock.

Reply:
@pcanal, thinking more about it, I think is safer to keep the lock as they are, keeping also the lock on `TVirtualFitter::GetFitter`, to avoid  having another thread calling `TVirtualFitter::SetFitter` at the same time. 
If you don't have objections I will then merge it as it is. Please let me know

Reply:
> I think is safer to keep the lock as they are, keeping also the lock on TVirtualFitter::GetFitter, to avoid having another thread calling TVirtualFitter::SetFitter at the same time.

My proposal was doing exactly that by moving the code line around.

> TVirtualFitter::GetFitter is a legacy functionality and in general should not be used, in particular in an application running MT.

Could this code be updated to use the thread safe alternative/newer-replacement?

Reply:
This code exists to maintain backward compatibility. What we can do is document better or deprecate `TVirtualFitter` to avoid using it in MT applications